### ISSUE TITLE
Bug: wrong BPM for "If you're not in it for love"

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#884]
Registers as 112 BPM, when it's really 125BPM.
Maybe we need a Right-click context menu > "Override BPM for this song" (see #195)

**Comment by mpogue2** [comment]
Also, right now we just look at the region: 60s - 70s to determine BPM, and range is fixed at 125±15BPM.
We can probably do better than that:
- Make range for patter/singer/vocals: 125 ± 8 BPM (99% of all square dance recordings)
- Make range for anything else: 125 ± 15 BPM
- If we get an answer, go with it.
- If we get an "out of range", then open up to 125 ± 15 BPM
- If we get an answer, go with it.
- Else return 0 (could not determine BPM, use percent instead).

Something like that is probably better than what we have now, and this will minimally impact load time (only 1% of the songs will hit the second MiniBPM call).

**Comment by mpogue2** [comment]
Fixed by: 5c8b1a4d148f4e7b9fee8dd8fdbd7022b2f36743 .

There were a bunch of songs that were generating the wrong BPM.  I factored out the BPM code, and did some tests by voting between various start and end points.  The winner was just to sample between 60 and 90 seconds into the song.  It worked in all but 1 of my 15 problematic cases.  That's probably as close as we'll get with this algorithm (miniBPM).

The only song that isn't detected correctly in my patter folder is now "Turnpike Warrior_Mom_Kandahar (Perpetual e-Motion)".
